plugins/gen-design-doc/aac_gen_design_doc/aac_gen_design_doc_impl.py

```diff
@@ -0,0 +1,103 @@
+"""AaC Plugin implementation module for the aac-gen-design-doc plugin."""
+
+import os
+import sys
+
+from iteration_utilities import flatten
+from jinja2 import Template
+
+from aac import parser, util
+from aac.template_engine import (
+    TemplateOutputFile,
+    generate_template,
+    load_templates,
+    write_generated_templates_to_file,
+)
+
+plugin_version = "0.0.1"
+default_template_file = "templates/system-design-doc.md.jinja2"
+
+
+def gen_design_doc(architecture_files: str, output_directory: str, template_file: str):
```

Comment:
```suggestion
def gen_design_doc(architecture_files: str, output_directory: str, template_file: str = None):
```

Reply:
I'd explicitly default it to None for readability since this is an optional command arg.